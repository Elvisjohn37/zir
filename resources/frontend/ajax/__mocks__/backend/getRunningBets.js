import { getError } from './errorData';
let { isFunction } = jest.requireActual('lodash');

let runningBetsData;
let isBlockSuccess = false;

function getRunningBets({ page, success, error }) {
	if (runningBetsData.data.result) {
		runningBetsData.data.data.content = runningBetsData.data.data.content.map((item) => {
			let finalData = {};

			for (let key in item) {
				finalData[key] = item[key] + page;
			}

			return finalData;
		});
		if (isFunction(success)) {
			!isBlockSuccess && success(runningBetsData);
		}
		return runningBetsData;
	} else {
		let errorData = getError();
		isFunction(error) && error(errorData);
		return errorData;
	}
}

function setRunningBetsData(result, data) {
	runningBetsData.data.result = result;
	runningBetsData.data.data = data;
}

function blockSuccess() {
	isBlockSuccess = true;
}

function getRunningBetsData() {
	return runningBetsData;
}

function resetRunningBetsData() {
	isBlockSuccess = false;
	runningBetsData = {
		data: {
			result: true,
			data: {
				content: [
					{
						event: 'R',
						betLinkFormat: 'noBetLink',
						reasonFormat: 'noReason',
						gameID: '00640',
						roundIDFormat: 'roundDetID',
						roundDetID: 1,
						tableNameFormat: 'gameName',
						gameName: 'Everything about Net (Mobile)',
						stake: '100.000000',
						turnover: '100.000000',
						dateTime: '2021-08-27 08:00:00.000000',
					},
					{
						event: 'R',
						betLinkFormat: 'noBetLink',
						reasonFormat: 'noReason',
						gameID: '01640',
						roundIDFormat: 'roundDetID',
						roundDetID: 2,
						tableNameFormat: 'gameName',
						gameName: 'Everything about Net (Desktop)',
						stake: '101.000000',
						turnover: '102.000000',
						dateTime: '2021-08-27 09:01:00.000000',
					},
					{
						event: 'R',
						betLinkFormat: 'noBetLink',
						reasonFormat: 'noReason',
						gameID: '01641',
						roundIDFormat: 'roundDetID',
						roundDetID: 2,
						tableNameFormat: 'gameName',
						gameName: 'Everything about Net (Desktop)',
						stake: '121.000000',
						turnover: '112.000000',
						dateTime: '2021-08-27 10:01:00.000000',
					},
				],
				totalRunningBets: '100.000000',
				totalRow: 1,
				rowsPerPage: 20,
			},
		},
	};
}
resetRunningBetsData();

module.exports = { getRunningBets, setRunningBetsData, resetRunningBetsData, getRunningBetsData, blockSuccess };
