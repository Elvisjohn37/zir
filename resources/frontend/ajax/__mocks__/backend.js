import * as getGame from './backend/getGame';
import * as errorData from './backend/errorData';
import * as getBalance from './backend/getBalance';
import * as getStatementReport from './backend/getStatementReport';
import * as getPlayerCreditList from './backend/getPlayerCreditList';
import * as getPlayerTransferList from './backend/getPlayerTransferList';
import * as getTransactionLog from './backend/getTransactionLog';
import * as getPlayerBetList from './backend/getPlayerBetList';
import * as getRunningBets from './backend/getRunningBets';
import * as getGameLedgerReport from './backend/getGameLedgerReport';
import * as getArcFinLedReport from './backend/getArcFinLedReport';
import * as getGameGuide from './backend/getGameGuide';
import * as getBanners from './backend/getBanners';
import * as getHomeConfig from './backend/getHomeConfig';

let mockedModule = jest.createMockFromModule('../backend.js');
let { isFunction } = jest.requireActual('lodash');

let getSessionData;
let getSessionResult;
mockedModule.getSession = (success, error) => {
	if (getSessionResult) {
		isFunction(success) && success(getSessionData);
		return getSessionData;
	} else {
		let getErrorData = errorData.getError();
		isFunction(error) && error(getErrorData);
		return getErrorData;
	}
};
mockedModule.setGetSession = (result, data) => {
	getSessionResult = result;
	getSessionData.data.data = data;
	getSessionData.data.result = getSessionResult;
};
mockedModule.resetGetSession = () => {
	getSessionResult = true;
	getSessionData = {
		data: { result: getSessionResult, data: { isLogin: true, userName: 'test' } },
	};
};
mockedModule.resetGetSession();

mockedModule.subscribe = function subscribe(event, eventFuntion) {
	if (event == 'error') {
		let getErrorData = errorData.getError();
		eventFuntion(getErrorData);
		return getErrorData;
	}
};

// from other modules
Object.assign(
	mockedModule,
	errorData,
	getGame,
	getBalance,
	getStatementReport,
	getPlayerCreditList,
	getPlayerTransferList,
	getPlayerBetList,
	getTransactionLog,
	getRunningBets,
	getGameLedgerReport,
	getArcFinLedReport,
	getGameGuide,
	getBanners,
	getHomeConfig
);

module.exports = mockedModule;
