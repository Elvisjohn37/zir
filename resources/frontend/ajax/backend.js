import axios from 'axios';
import { subscribe, unsubscribe, responseInterceptor } from './backend/responseInterceptor';
import { autoAddCsrfToken } from './backend/csrfToken';
import defaultRequestHeaders from './backend/defaultRequestHeaders';

let axiosInstance = axios.create({
	baseURL: process.env.APP_ROOT || '',
});

defaultRequestHeaders(axiosInstance);
autoAddCsrfToken(axiosInstance);
responseInterceptor(axiosInstance);

export { subscribe, unsubscribe };

export function getSession(success, error) {
	axiosInstance
		.get('/getsession')
		.then(success)
		.catch(error ? error : () => {});
}

export function getGame({ isMobileDevice, page, success, error, ...otherOptions }) {
	axiosInstance({
		url: '/getgame',
		method: 'post',
		data: {
			page,
			isMobileDevice,
		},
		...otherOptions,
	})
		.then(success)
		.catch(error);
}

export function getbsiGames({ isMobileDevice, page, success, error, ...otherOptions }) {
	axiosInstance({
		url: '/getbsiGames',
		method: 'post',
		data: {
			page,
			isMobileDevice,
		},
		...otherOptions,
	})
		.then(success)
		.catch(error);
}

export function getAnnouncement({ success, error, viewType, showConfig, ...otherOptions }) {
	axiosInstance({
		url: '/getannouncement',
		method: 'post',
		data: {
			viewType,
			showConfig,
		},
		...otherOptions,
	})
		.then(success)
		.catch(error ? error : () => {});
}

export function getBalance(success, error) {
	axiosInstance.post('/getbalance').then(success).catch(error);
}

export function getBalanceBackground(success, error) {
	axiosInstance.post('/getbalancebg').then(success).catch(error);
}

export function getGameGuide({ product, gameType, lang, success, error, ...otherOptions }) {
	axiosInstance({
		url: '/getgameguide',
		method: 'post',
		data: {
			product,
			gameType,
			lang,
		},
		...otherOptions,
	})
		.then(success)
		.catch(error ? error : () => {});
}

export function logout(success, error) {
	axiosInstance
		.post('/logout')
		.then(success)
		.catch(error ? error : () => {});
}

export function getRunningBets({ page, success, error }) {
	axiosInstance({
		url: '/getrunningbets',
		method: 'post',
		data: {
			page,
		},
	})
		.then(success)
		.catch(error ? error : () => {});
}

export function getStatementReport(rangeNo, showConfig, success, error) {
	axiosInstance({
		url: '/getstatementreport',
		method: 'post',
		data: {
			rangeNo,
			showConfig,
		},
	})
		.then(success)
		.catch(error ? error : () => {});
}

export function checkIsMaintenance(success, error) {
	axiosInstance
		.get('/ismaintenance')
		.then(success)
		.catch(error ? error : () => {});
}

export function getPlayerBetList({ date, productID, page, success, error }) {
	axiosInstance({
		url: '/getplayerbetList',
		method: 'post',
		data: {
			date,
			page,
			productID,
		},
	})
		.then(success)
		.catch(error ? error : () => {});
}

export function getPlayerPromoList({ date, page, success, error }) {
	axiosInstance({
		url: '/getplayerpromolist',
		method: 'post',
		data: {
			date,
			page,
		},
	})
		.then(success)
		.catch(error ? error : () => {});
}

export function getPlayerTransferList({ date, page, success, error }) {
	axiosInstance({
		url: '/getplayertransferlist',
		method: 'post',
		data: {
			date,
			page,
		},
	})
		.then(success)
		.catch(error ? error : () => {});
}

export function getPlayerCreditList({ date, page, success, error }) {
	axiosInstance({
		url: '/getplayercreditlist',
		method: 'post',
		data: { date, page },
	})
		.then(success)
		.catch(error ? error : () => {});
}

export function getLayoutConfig(type, success, error) {
	let url;

	switch (type) {
		case 'main':
			url = '/getmainlayoutconfig';
			break;
		case 'noLayout':
			url = '/getnolayoutconfig';
	}

	axiosInstance
		.post(url)
		.then(success)
		.catch(error ? error : () => {});
}

export function getTransactionLog({ date, page, showConfig, success, error }) {
	axiosInstance({
		url: '/gettransactionlog',
		method: 'post',
		data: {
			date,
			page,
			showConfig,
		},
	})
		.then(success)
		.catch(error ? error : () => {});
}

export function getGameLedgerReport({ data, success, error }) {
	axiosInstance({
		url: '/getgameledger',
		method: 'post',
		data: {
			date: data.date,
			page: data.page,
			showConfig: data.showConfig,
		},
	})
		.then(success)
		.catch(error ? error : () => {});
}

export function getArcFinLedReport({ data, success, error }) {
	axiosInstance({
		url: '/getarcfinledreport',
		method: 'post',
		data: {
			date: data.date,
			page: data.page,
			showConfig: data.showConfig,
		},
	})
		.then(success)
		.catch(error ? error : () => {});
}

export function getBanners({ isMobileDevice, type, success, error }) {
	axiosInstance({
		url: '/getbanners',
		method: 'post',
		data: {
			type: type,
			isMobileDevice: isMobileDevice,
		},
	})
		.then(success)
		.catch(error ? error : () => {});
}

export function settleRunningGame({ success, error }) {
	axiosInstance({
		url: '/settlerunninggame',
		method: 'post',
	})
		.then(success)
		.catch(error ? error : () => {});
}

export function getHomeConfig({ type, success, error }) {
	axiosInstance({
		url: '/gethomeconfig',
		method: 'post',
		data: {
			type: type,
		},
	})
		.then(success)
		.catch(error ? error : () => {});
}

export function getBsiGames({ page, isMobileDevice, success, error }) {
	axiosInstance({
		url: '/getBsiGames',
		method: 'post',
		data: {
			page,
			isMobileDevice,
		},
	})
		.then(success)
		.catch(error ? error : () => {});
}

export function getGameCategories(page, success, error) {
	axiosInstance({
		url: '/getgamecategories',
		method: 'post',
		data: {
			page: page,
		},
	})
		.then(success)
		.catch(error ? error : () => {});
}
