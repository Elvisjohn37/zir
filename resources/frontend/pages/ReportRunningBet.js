import React, { useEffect } from 'react';
import styles from './ReportRunningBet.module';
import { LangDataProvider } from 'frontend/components/Language';
import Sidebar from 'frontend/components/Sidebar';
import { FormattedMessage } from 'react-intl';
import TableIndicator from 'frontend/components/TableIndicator';
import { getRunningBets } from 'frontend/ajax/backend';
import MoneyFormat from 'frontend/components/MoneyFormat';
import Skeleton from 'frontend/components/Skeleton';
import Table from '@mui/material/Table';
import TableHead from '@mui/material/TableHead';
import TableRow from '@mui/material/TableRow';
import TableBody from '@mui/material/TableBody';
import TableCell from '@mui/material/TableCell';
import TableContainer from '@mui/material/TableContainer';
import MediaQuery from 'react-responsive';
import { MobileTableRefresh, DesktopTableRefresh } from 'frontend/components/TableRefresh';
import { DateTime } from 'frontend/components/ReportCommonFormatter';
import { GameDetails, GameResult, StakeTurnover } from 'frontend/components/ReportBetFormatter';
import classnames from 'classnames';
import { LayoutConsumer } from 'frontend/contexts/Layout';
import Spinner from 'frontend/components/Spinner';

export default function ReportRunningBet() {
	let [runningBets, setRunningBets] = React.useState({
		content: [],
		totalRunningBets: 0,
		totalRow: [],
		rowsPerPage: 0,
	});
	let [totalRunningBetsError, setTotalRunningBetsError] = React.useState(false);
	let [activePage, setActivePage] = React.useState(0);

	let [isFetching, setIsFetching] = React.useState(true);

	useEffect(() => {
		if (isFetching) {
			setIsFetching(true);
			getRunningBets({
				page: activePage + 1,
				success: (response) => {
					setIsFetching(false);

					setRunningBets(response.data.data);
				},
				error: () => {
					setTotalRunningBetsError(true);
					setIsFetching(false);
				},
			});
		}
	}, [isFetching]);

	return (
		<LayoutConsumer>
			{({ layoutState }) => (
				<LangDataProvider category="reports" fallback={<Spinner />}>
					<Sidebar sidebarGroup="archivedReport" disabled={layoutState.config.isMobileSite}>
						<div className={styles.tableSection}>
							<MediaQuery maxDeviceWidth={styles.main_mobile}>
								{(matches) =>
									matches ? (
										<MobileTableRefresh
											onClick={() => setIsFetching(true)}
											isDisabled={isFetching}
											display={<FormattedMessage id="runningBets" />}
											hasBackButton={false}
										/>
									) : (
										<DesktopTableRefresh
											onClick={() => setIsFetching(true)}
											isDisabled={isFetching}
											id="runningBetRefresh"
											values={{ product: <FormattedMessage id="runningBets" /> }}
											hasBackButton={false}
										/>
									)
								}
							</MediaQuery>
							<div className={classnames(styles.totalRunningBets)}>
								<FormattedMessage
									id="totalRunningBets"
									values={{
										amount: isFetching ? (
											<Skeleton height={10} width={30} className={styles.loader} />
										) : (
											<MoneyFormat value={runningBets.totalRunningBets} />
										),
									}}
								/>
							</div>
							<div>
								<TableIndicator
									id="testTableInit"
									isLoading={isFetching}
									isEmpty={runningBets.content.length == 0}
									isError={totalRunningBetsError}
									pagination={{
										count: runningBets.totalRow,
										page: activePage,
										rowsPerPage: runningBets.rowsPerPage,
										onPageChange: (event, page) => {
											setActivePage(page);
											setIsFetching(true);
										},
									}}
								>
									<TableContainer>
										<Table classes={{ root: styles.table }}>
											<TableHead>
												<TableRow classes={{ root: styles.row }}>
													<TableCell align="center" className={styles.colHeaderRowNum}>
														#
													</TableCell>
													<TableCell align="center" className={styles.colHeaderDateTime}>
														<FormattedMessage id="date" /> &amp;{' '}
														<FormattedMessage id="time" />
													</TableCell>
													<TableCell align="center" className={styles.colGameDetails}>
														<FormattedMessage id="gameDetails" />
													</TableCell>
													<TableCell align="center" className={styles.colHeaderTurnover}>
														<FormattedMessage id="stakeTurnover" />
													</TableCell>
													<TableCell align="center" className={styles.colstatus}>
														<FormattedMessage id="status" />
													</TableCell>
												</TableRow>
											</TableHead>

											<TableBody>
												{runningBets.content.map((row, index) => (
													<TableRow key={runningBets.rowsPerPage * activePage + index + 1}>
														<TableCell align="left" scope="row">
															{runningBets.rowsPerPage * activePage + index + 1}
														</TableCell>
														<TableCell align="center">
															<DateTime dateTime={row.dateTime} newLine={true} />
														</TableCell>
														<TableCell align="center">
															<GameDetails row={row} />
														</TableCell>
														<TableCell align="right">
															<StakeTurnover row={row} />
														</TableCell>
														<TableCell align="center">
															<GameResult row={row} />
														</TableCell>
													</TableRow>
												))}
											</TableBody>
										</Table>
									</TableContainer>
								</TableIndicator>
							</div>
						</div>
					</Sidebar>
				</LangDataProvider>
			)}
		</LayoutConsumer>
	);
}
