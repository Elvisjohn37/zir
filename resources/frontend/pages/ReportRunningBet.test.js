import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import ReportRunningBet from './ReportRunningBet';
import * as MediaQuery from 'react-responsive';
import { setUserState } from 'frontend/contexts/User';
import userEvent from '@testing-library/user-event';
import { blockSuccess } from 'frontend/ajax/__mocks__/backend/getRunningBets';
import * as runningBets from 'frontend/ajax/backend';

jest.mock('@mui/material/Table');
jest.mock('@mui/material/TableHead');
jest.mock('@mui/material/TableRow');
jest.mock('@mui/material/TableBody');
jest.mock('@mui/material/TableCell');
jest.mock('@mui/material/TableContainer');
jest.mock('frontend/components/TableIndicator');
jest.mock('frontend/components/Sidebar');
jest.mock('frontend/components/Language');
jest.mock('frontend/components/ReportCommonFormatter');
jest.mock('frontend/components/ReportBetFormatter');
jest.mock('frontend/components/TableRefresh');
jest.mock('frontend/components/MoneyFormat');
jest.mock('frontend/components/Skeleton');
jest.mock('frontend/contexts/Layout');
jest.mock('frontend/contexts/User');
jest.mock('react-responsive');
jest.mock('react-intl');
jest.mock('frontend/ajax/backend');

describe('<ReportRunningBet/>', () => {
	beforeEach(() => {
		jest.spyOn(runningBets, 'getRunningBets');
		jest.spyOn(MediaQuery, 'setMediaQuery');
	});

	afterEach(() => {
		resetMocks();
		cleanup();
	});

	function resetMocks() {
		jest.clearAllMocks();
		jest.restoreAllMocks();
		runningBets.resetRunningBetsData();
		MediaQuery.setMediaQuery(true);
	}

	function expectCorrectTableContent() {
		setUserState({ user: { parent: 'test' } });
		let responseData = runningBets.getRunningBetsData().data.data;
		let columnsLength = 5;
		responseData.content.forEach((data, index) => {
			expect(screen.getAllByTestId('TableCell')[columnsLength * index + columnsLength]).toBeInTheDocument();
			expect(screen.getAllByTestId('TableCell')[columnsLength * index + columnsLength + 1]).toBeInTheDocument();
			expect(screen.getAllByTestId('TableCell')[columnsLength * index + columnsLength + 2]).toBeInTheDocument();
			expect(screen.getAllByTestId('TableCell')[columnsLength * index + columnsLength + 3]).toBeInTheDocument();
			expect(screen.getAllByTestId('TableCell')[columnsLength * index + columnsLength + 4]).toBeInTheDocument();
		});
	}

	it('should match snapshot', () => {
		let { asFragment } = render(<ReportRunningBet />);
		expect(asFragment()).toMatchSnapshot();
	});

	it('should have table loading indicator on table during data fetching', () => {
		blockSuccess();
		render(<ReportRunningBet />);
		expect(screen.getByText('Loading TableIndicator')).toBeInTheDocument();
	});

	it('should fetch the latest running bet record from backend', () => {
		render(<ReportRunningBet />);
		expect(runningBets.getRunningBets).toHaveBeenCalledWith({
			page: 1,
			success: expect.any(Function),
			error: expect.any(Function),
		});
		expectCorrectTableContent();
	});

	it('should update and fetch latest record when refresh button is clicked', () => {
		render(<ReportRunningBet />);
		expectCorrectTableContent();
		userEvent.click(screen.getByTestId('MobileTableRefresh'));
		expect(runningBets.getRunningBets).toHaveBeenNthCalledWith(1, {
			page: 1,
			success: expect.any(Function),
			error: expect.any(Function),
		});

		runningBets.setRunningBetsData(true, {
			content: [
				{
					event: 'R',
					betLinkFormat: 'noBetLink',
					reasonFormat: 'noReason',
					gameID: '00641',
					roundIDFormat: 'roundDetID',
					roundDetID: 2,
					tableNameFormat: 'gameName',
					gameName: 'Dangdut Queen (Mobile)',
					stake: '100.000000',
					turnover: '100.000000',
					dateTime: '2021-08-26 08:00:00.000000',
				},
				{
					event: 'R',
					betLinkFormat: 'noBetLink',
					reasonFormat: 'noReason',
					gameID: '01641',
					roundIDFormat: 'roundDetID',
					roundDetID: 4,
					tableNameFormat: 'gameName',
					gameName: 'Dangdut Queen (Desktop)',
					stake: '111.000000',
					turnover: '102.000000',
					dateTime: '2021-08-28 09:01:00.000000',
				},
				{
					event: 'R',
					betLinkFormat: 'noBetLink',
					reasonFormat: 'noReason',
					gameID: '01643',
					roundIDFormat: 'roundDetID',
					roundDetID: 6,
					tableNameFormat: 'gameName',
					gameName: 'Dangdut Queen (Desktop)',
					stake: '111.000000',
					turnover: '112.000000',
					dateTime: '2021-08-28 10:01:00.000000',
				},
			],
			totalRunningBets: '100.000000',
			totalRow: 1,
			rowsPerPage: 20,
		});
		expect(runningBets.getRunningBets).toHaveBeenNthCalledWith(2, {
			page: 1,
			success: expect.any(Function),
			error: expect.any(Function),
		});
	});

	it('should display pagination and record base on selected page', () => {
		render(<ReportRunningBet />);
		expect(screen.getByText('Pagination count: 1')).toBeInTheDocument();
		expect(screen.getByText('Pagination page: 0')).toBeInTheDocument();
		expect(screen.getByText('Pagination rowsPerPage: 20')).toBeInTheDocument();
		userEvent.click(screen.getByTestId('TableIndicator'));
		expect(runningBets.getRunningBets).toHaveBeenNthCalledWith(2, {
			page: 2,
			success: expect.any(Function),
			error: expect.any(Function),
		});
	});

	it('should display no record found when record is empty', () => {
		runningBets.setRunningBetsData(true, { content: [] });
		let { container } = render(<ReportRunningBet />);
		let tableRow = container.getElementsByClassName('testTableRow');
		expect(tableRow).toHaveLength(0);
	});

	it('should display error message when backend response is not valid', () => {
		runningBets.setRunningBetsData(false);
		render(<ReportRunningBet />);
		expect(screen.getByText('Error Response')).toBeInTheDocument();
	});
});
