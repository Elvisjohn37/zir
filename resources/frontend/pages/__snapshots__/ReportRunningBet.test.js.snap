// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ReportRunningBet/> should match snapshot 1`] = `
<DocumentFragment>
  <div
    category="reports"
    data-testid="LangDataProvider"
  >
    <div>
      reports
    </div>
    <div
      data-testid="default"
      sidebargroup="report"
    >
      <div
        class="tableSection"
      >
        <div
          data-testid="MobileTableRefresh"
          hasbackbutton="false"
          isdisabled="false"
        />
        <div>
          <div
            data-testid="FormattedMessage"
            id="totalRunningBets"
            values="[object Object]"
          >
            <div>
              totalRunningBets
            </div>
          </div>
        </div>
        <div>
          <div
            data-testid="TableIndicator"
            id="testTableInit"
            isempty="false"
            iserror="false"
            isloading="false"
            pagination="{\\"count\\":1,\\"page\\":0,\\"rowsPerPage\\":20}"
          >
            <span>
              Pagination count: 1
            </span>
            <span>
              Pagination page: 0
            </span>
            <span>
              Pagination rowsPerPage: 20
            </span>
            <span>
              Done loading TableIndicator
            </span>
            <div
              data-testid="TableContainer"
            >
              <div
                aria-label="a dense table"
                classes="{\\"root\\":\\"table\\"}"
                data-testid="Table"
              >
                <div
                  data-testid="TableHead"
                >
                  <div
                    classes="{\\"root\\":\\"row\\"}"
                    data-testid="TableRow"
                  >
                    <div
                      align="center"
                      class="colHeaderRowNum"
                      data-testid="TableCell"
                    >
                      #
                    </div>
                    <div
                      align="center"
                      class="colHeaderDateTime"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="FormattedMessage"
                        id="date"
                      >
                        <div>
                          date
                        </div>
                      </div>
                       & 
                      <div
                        data-testid="FormattedMessage"
                        id="time"
                      >
                        <div>
                          time
                        </div>
                      </div>
                    </div>
                    <div
                      align="center"
                      class="colGameDetails"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="FormattedMessage"
                        id="gameDetails"
                      >
                        <div>
                          gameDetails
                        </div>
                      </div>
                    </div>
                    <div
                      align="center"
                      class="colHeaderTurnover"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="FormattedMessage"
                        id="stakeTurnover"
                      >
                        <div>
                          stakeTurnover
                        </div>
                      </div>
                    </div>
                    <div
                      align="center"
                      class="colstatus"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="FormattedMessage"
                        id="status"
                      >
                        <div>
                          status
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  data-testid="TableBody"
                >
                  <div
                    data-testid="TableRow"
                  >
                    <div
                      align="left"
                      data-testid="TableCell"
                      scope="row"
                    >
                      1
                    </div>
                    <div
                      align="center"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="DateTime"
                      />
                    </div>
                    <div
                      align="center"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="GameDetails"
                        row="{\\"event\\":\\"R1\\",\\"betLinkFormat\\":\\"noBetLink1\\",\\"reasonFormat\\":\\"noReason1\\",\\"gameID\\":\\"006401\\",\\"roundIDFormat\\":\\"roundDetID1\\",\\"roundDetID\\":2,\\"tableNameFormat\\":\\"gameName1\\",\\"gameName\\":\\"Everything about Net (Mobile)1\\",\\"stake\\":\\"100.0000001\\",\\"turnover\\":\\"100.0000001\\",\\"dateTime\\":\\"2021-08-27 08:00:00.0000001\\"}"
                      >
                        <span>
                          {"event":"R1","betLinkFormat":"noBetLink1","reasonFormat":"noReason1","gameID":"006401","roundIDFormat":"roundDetID1","roundDetID":2,"tableNameFormat":"gameName1","gameName":"Everything about Net (Mobile)1","stake":"100.0000001","turnover":"100.0000001","dateTime":"2021-08-27 08:00:00.0000001"}
                        </span>
                      </div>
                    </div>
                    <div
                      align="right"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="StakeTurnover"
                        row="{\\"event\\":\\"R1\\",\\"betLinkFormat\\":\\"noBetLink1\\",\\"reasonFormat\\":\\"noReason1\\",\\"gameID\\":\\"006401\\",\\"roundIDFormat\\":\\"roundDetID1\\",\\"roundDetID\\":2,\\"tableNameFormat\\":\\"gameName1\\",\\"gameName\\":\\"Everything about Net (Mobile)1\\",\\"stake\\":\\"100.0000001\\",\\"turnover\\":\\"100.0000001\\",\\"dateTime\\":\\"2021-08-27 08:00:00.0000001\\"}"
                      >
                        <div>
                          100.0000001
                        </div>
                        <div>
                          100.0000001
                        </div>
                      </div>
                    </div>
                    <div
                      align="center"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="GameResult"
                        row="{\\"event\\":\\"R1\\",\\"betLinkFormat\\":\\"noBetLink1\\",\\"reasonFormat\\":\\"noReason1\\",\\"gameID\\":\\"006401\\",\\"roundIDFormat\\":\\"roundDetID1\\",\\"roundDetID\\":2,\\"tableNameFormat\\":\\"gameName1\\",\\"gameName\\":\\"Everything about Net (Mobile)1\\",\\"stake\\":\\"100.0000001\\",\\"turnover\\":\\"100.0000001\\",\\"dateTime\\":\\"2021-08-27 08:00:00.0000001\\"}"
                      />
                    </div>
                  </div>
                  <div
                    data-testid="TableRow"
                  >
                    <div
                      align="left"
                      data-testid="TableCell"
                      scope="row"
                    >
                      2
                    </div>
                    <div
                      align="center"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="DateTime"
                      />
                    </div>
                    <div
                      align="center"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="GameDetails"
                        row="{\\"event\\":\\"R1\\",\\"betLinkFormat\\":\\"noBetLink1\\",\\"reasonFormat\\":\\"noReason1\\",\\"gameID\\":\\"016401\\",\\"roundIDFormat\\":\\"roundDetID1\\",\\"roundDetID\\":3,\\"tableNameFormat\\":\\"gameName1\\",\\"gameName\\":\\"Everything about Net (Desktop)1\\",\\"stake\\":\\"101.0000001\\",\\"turnover\\":\\"102.0000001\\",\\"dateTime\\":\\"2021-08-27 09:01:00.0000001\\"}"
                      >
                        <span>
                          {"event":"R1","betLinkFormat":"noBetLink1","reasonFormat":"noReason1","gameID":"016401","roundIDFormat":"roundDetID1","roundDetID":3,"tableNameFormat":"gameName1","gameName":"Everything about Net (Desktop)1","stake":"101.0000001","turnover":"102.0000001","dateTime":"2021-08-27 09:01:00.0000001"}
                        </span>
                      </div>
                    </div>
                    <div
                      align="right"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="StakeTurnover"
                        row="{\\"event\\":\\"R1\\",\\"betLinkFormat\\":\\"noBetLink1\\",\\"reasonFormat\\":\\"noReason1\\",\\"gameID\\":\\"016401\\",\\"roundIDFormat\\":\\"roundDetID1\\",\\"roundDetID\\":3,\\"tableNameFormat\\":\\"gameName1\\",\\"gameName\\":\\"Everything about Net (Desktop)1\\",\\"stake\\":\\"101.0000001\\",\\"turnover\\":\\"102.0000001\\",\\"dateTime\\":\\"2021-08-27 09:01:00.0000001\\"}"
                      >
                        <div>
                          101.0000001
                        </div>
                        <div>
                          102.0000001
                        </div>
                      </div>
                    </div>
                    <div
                      align="center"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="GameResult"
                        row="{\\"event\\":\\"R1\\",\\"betLinkFormat\\":\\"noBetLink1\\",\\"reasonFormat\\":\\"noReason1\\",\\"gameID\\":\\"016401\\",\\"roundIDFormat\\":\\"roundDetID1\\",\\"roundDetID\\":3,\\"tableNameFormat\\":\\"gameName1\\",\\"gameName\\":\\"Everything about Net (Desktop)1\\",\\"stake\\":\\"101.0000001\\",\\"turnover\\":\\"102.0000001\\",\\"dateTime\\":\\"2021-08-27 09:01:00.0000001\\"}"
                      />
                    </div>
                  </div>
                  <div
                    data-testid="TableRow"
                  >
                    <div
                      align="left"
                      data-testid="TableCell"
                      scope="row"
                    >
                      3
                    </div>
                    <div
                      align="center"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="DateTime"
                      />
                    </div>
                    <div
                      align="center"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="GameDetails"
                        row="{\\"event\\":\\"R1\\",\\"betLinkFormat\\":\\"noBetLink1\\",\\"reasonFormat\\":\\"noReason1\\",\\"gameID\\":\\"016411\\",\\"roundIDFormat\\":\\"roundDetID1\\",\\"roundDetID\\":3,\\"tableNameFormat\\":\\"gameName1\\",\\"gameName\\":\\"Everything about Net (Desktop)1\\",\\"stake\\":\\"121.0000001\\",\\"turnover\\":\\"112.0000001\\",\\"dateTime\\":\\"2021-08-27 10:01:00.0000001\\"}"
                      >
                        <span>
                          {"event":"R1","betLinkFormat":"noBetLink1","reasonFormat":"noReason1","gameID":"016411","roundIDFormat":"roundDetID1","roundDetID":3,"tableNameFormat":"gameName1","gameName":"Everything about Net (Desktop)1","stake":"121.0000001","turnover":"112.0000001","dateTime":"2021-08-27 10:01:00.0000001"}
                        </span>
                      </div>
                    </div>
                    <div
                      align="right"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="StakeTurnover"
                        row="{\\"event\\":\\"R1\\",\\"betLinkFormat\\":\\"noBetLink1\\",\\"reasonFormat\\":\\"noReason1\\",\\"gameID\\":\\"016411\\",\\"roundIDFormat\\":\\"roundDetID1\\",\\"roundDetID\\":3,\\"tableNameFormat\\":\\"gameName1\\",\\"gameName\\":\\"Everything about Net (Desktop)1\\",\\"stake\\":\\"121.0000001\\",\\"turnover\\":\\"112.0000001\\",\\"dateTime\\":\\"2021-08-27 10:01:00.0000001\\"}"
                      >
                        <div>
                          121.0000001
                        </div>
                        <div>
                          112.0000001
                        </div>
                      </div>
                    </div>
                    <div
                      align="center"
                      data-testid="TableCell"
                    >
                      <div
                        data-testid="GameResult"
                        row="{\\"event\\":\\"R1\\",\\"betLinkFormat\\":\\"noBetLink1\\",\\"reasonFormat\\":\\"noReason1\\",\\"gameID\\":\\"016411\\",\\"roundIDFormat\\":\\"roundDetID1\\",\\"roundDetID\\":3,\\"tableNameFormat\\":\\"gameName1\\",\\"gameName\\":\\"Everything about Net (Desktop)1\\",\\"stake\\":\\"121.0000001\\",\\"turnover\\":\\"112.0000001\\",\\"dateTime\\":\\"2021-08-27 10:01:00.0000001\\"}"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
